src/main/java/com/synopsys/integration/detect/workflow/componentlocationanalysis/GenerateComponentLocationAnalysisOperation.java

```diff
@@ -1,69 +1,105 @@
+        invokeComponentLocator(componentsList, scanOutputFolder, projectSrcDir);
     }
 
-    public static File locateComponentsForOfflineDetectorScan(BdioResult bdio, File scanOutputFolder) throws DetectUserFriendlyException {
-        // In Part II:
-            // given a BDIO, call BdioToCLLComponentTransformer to get CLL input file (components-source.json)
-            // call library w/ CLL input
-            // return the resulting file (not anticipating any post-processing except saving it in the correct directory)
-        return generatePlaceHolderJsonFileForNow(scanOutputFolder);
+    /**
+     * Given a BDIO, generates an output file consisting of the list of unique components detected and their declaration
+     * locations.
+     * @param bdio from running offline Detector scan
+     * @param scanOutputFolder Detect's output subdirectory where this file will be saved
+     * @param projectSrcDir source directory of project being scanned
+     * @throws DetectUserFriendlyException
+     */
+    public void locateComponentsForOfflineDetectorScan(BdioResult bdio, File scanOutputFolder, File projectSrcDir) {
+        List<Component> componentsList = bdioTransformer.transformBdioToComponentList(bdio);
+        invokeComponentLocator(componentsList, scanOutputFolder, projectSrcDir);
+    }
+
+    private void invokeComponentLocator(List<Component> componentsList, File scanOutputFolder, File projectSrcDir) {
+        Input componentLocatorInput = generateComponentLocatorInput(componentsList, projectSrcDir);
+        ///////////////////////////////////////TODO remove next 3 lines before merging to master//////////////////////////////////
+        try {
+            serializeInputToJson(scanOutputFolder, componentLocatorInput);
+        } catch (DetectUserFriendlyException e) { logger.debug("Failed to save input file to CLL");}
+        ////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+        try {
+            int status = Application.locateComponents(componentLocatorInput, scanOutputFolder.toString() + "/" + DETECT_OUTPUT_FILE_NAME);
+            if (status != 0) {
+                logger.info("Component Locator execution was unsuccessful. Failed to generate Component Location Analysis file.");
+            }
+        } catch (Exception e) {
+            logger.info("There was a problem during Component Locator execution. Failed to generate Component Location Analysis file.");
+        }
+        callOnComponentLocatorToLocateComponents(componentLocatorInput, scanOutputFolder);
+    }
+
+    private Input generateComponentLocatorInput(List<Component> componentsList, File sourceDir) {
+        return new Input(sourceDir.getAbsolutePath(), new JsonObject(), componentsList);
+    }
+
+    private void callOnComponentLocatorToLocateComponents(Input componentLocatorInputObj, File scanOutputFolder) {
```

Comment:
Nit: refactor to a verb name convention -> callComponentLocatorToLocateComponents ([ref](https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html))

Reply:
Or just runComponentLocator.